internal/controller/resource_manager.go

```diff
@@ -278,3 +314,71 @@ func removeListSuffix(s string) string {
 
 	return s
 }
+
+// newWatcher creates a new Kubernetes watcher starting from the given resource version.
+func (r *ResourceManager) newWatcher(ctx context.Context, apiGroup string, resourceVersion string, existing watch.Interface, logger *zap.Logger) (watch.Interface, error) {
+	// Stop any existing watcher before creating a new one.
+	if existing != nil {
+		existing.Stop()
+	}
```

Comment:
1. `existing` is always set to `nil`. Remove that parameter altogether.
1. Remove `apiGroup`. This method must now use the new `r.apiGroup` field instead.
```suggestion
// newWatcher creates a new Kubernetes watcher starting from the given resource version.
func (r *ResourceManager) newWatcher(ctx context.Context, resourceVersion string, logger *zap.Logger) (watch.Interface, error) {
```

Reply:
addressed